paddle/phi/kernels/funcs/CMakeLists.txt

```diff
@@ -5,6 +5,8 @@ add_subdirectory(detail)
 
 math_library(deformable_conv_functor DEPS dense_tensor)
 math_library(concat_and_split_functor DEPS dense_tensor)
+target_compile_definitions(concat_and_split_functor
```

Comment:
这部分是不同版本的MSVC编译器对aligned storage的区别，相关Issue可以看  https://github.com/ceres-solver/ceres-solver/issues/481

```text
Use #define _ENABLE_EXTENDED_ALIGNED_STORAGE
This would make the alignment work correctly but has the drawback, that the compiled library is no longer ABI compatible to code compiled with an older version of MSVC.
Use #define _DISABLE_EXTENDED_ALIGNED_STORAGE
This would use the wrong alignment but mixing the code with code compiled from a different version would work.
```
1. 开启 _ENABLE_EXTENDED_ALIGNED_STORAGE 就会使用正确的align type，但是不兼容旧MSVC
2. 开启 _DISABLE_EXTENDED_ALIGNED_STORAGE，则会使用错误的aligntype，但是能保证兼容

综上还是保证正确的align type，开启 _ENABLE_EXTENDED_ALIGNED_STORAGE